debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java

```diff
@@ -161,32 +152,71 @@ else if (slotInfo.active()) {
                 }
             });
 
-            if (shouldCreateSlot || !slotInfo.hasValidFlushedLsn()) {
+            if (slotCreationInfo != null) {
+                this.defaultStartingPos = slotCreationInfo.startLsn();
+            }
+            else if (shouldCreateSlot || !slotInfo.hasValidFlushedLsn()) {
                 // this is a new slot or we weren't able to read a valid flush LSN pos, so we always start from the xlog pos that was reported
                 this.defaultStartingPos = xlogStart.get();
-            } else {
+            }
+            else {
                 Long latestFlushedLsn = slotInfo.latestFlushedLsn();
                 this.defaultStartingPos = latestFlushedLsn < xlogStart.get() ? latestFlushedLsn : xlogStart.get();
                 if (LOGGER.isDebugEnabled()) {
                     LOGGER.debug("found previous flushed LSN '{}'", ReplicationConnection.format(latestFlushedLsn));
                 }
             }
+            hasInitedSlot = true;
         } catch (SQLException e) {
             throw new JdbcConnectionException(e);
         }
     }
 
     private boolean useTemporarySlot() throws SQLException {
-        return dropSlotOnClose && pgConnection().getServerMajorVersion() >= 10;
+        return dropSlotOnClose && isPg10OrGreater();
+    }
+
+    private boolean exportSnapshotOnCreate() throws SQLException {
+        return exportSnapshot && isPg10OrGreater();
+    }
+
+    private boolean isPg10OrGreater() throws SQLException {
+        return pgConnection().getServerMajorVersion() >= 10;
     }
 
+    /**
+     * creating a replication connection and starting to stream involves a few steps:
+     * 1. we create the connection and ensure that
+     * a. the slot exists
+     * b. the slot isn't currently being used
+     * 2. we query to get our potential start position in the slot (lsn)
+     * 3. we try and start streaming, depending on our options (such as in wal2json)
+     *    this may fail, which can result in the connection being killed and we need to start
+     *    the process over if we are using a temporary slot
+     * 4. actually start the streamer
+     *
+     * This method takes care of all of these and this method queries for a default starting position
+     * If you know where you are starting from you should call {@link #startStreaming(Long)}, this method
+     * delegates to that method
+     * @return
+     * @throws SQLException
+     * @throws InterruptedException
+     */
     @Override
     public ReplicationStream startStreaming() throws SQLException, InterruptedException {
+        // this ensures we have the right type of connection, creates the slot if needed,
+        // and sets the defaultStartingPos, so we must set it here
+        initConnection();
```

Comment:
Seems superfluous as it's done again at line 218.

Reply:
this is needed because `defaultStartingPos` is set by calling initConnection. The comment there should highlight it better. We then also have it do in the in the `startStreaming(Long offset)` because both of these methods are called in the app.

Reply:
@addisonj, it seems the same effect can be achieved by simply calling `return startStreaming(null);` then? The `offset` will then be set to `defaultStartingPos`.